mozilla-release/browser/base/content/browser.js

```diff
@@ -7396,6 +7399,44 @@ var gRemoteTabsUI = {
   }
 };
 
+// Small utility handler which propagates current tab "privateness" property as
+// available to CSS selectors through #navigator-toolbox[on_private_tab]
+const _gPrivateTabUI = {
+  init: function PrivateTabUI_init() {
```

Comment:
In this particular case I'd find it inappropriate.
We need a corresponding "destructor" called at the right moment, and having `new X()` + `deinit` looks less obvious to me than a pair of `init` - `deinit`.